src/runtime_src/core/tools/xbmgmt2/OO_UpdateBase.cpp

```diff
@@ -0,0 +1,755 @@
+#include "tools/common/XBUtilitiesCore.h"
+namespace XBU = XBUtilities;
+
+// 3rd Party Library - Include Files
+#include <boost/filesystem.hpp>
+#include <boost/format.hpp>
+#include <boost/program_options.hpp>
+#include <boost/tokenizer.hpp>
+namespace po = boost::program_options;
+
+// System - Include Files
+#include <atomic>
+#include <chrono>
+#include <ctime>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <thread>
+
+#ifdef _WIN32
+#pragma warning(disable : 4996) // disable warning caused by std::asctime
+#endif
+
+OO_UpdateBase::OO_UpdateBase(const std::string& _longName, const std::string& _shortName, bool _isHidden)
+    : OptionOptions(_longName,
+                    _shortName,
+                    "Update base partition",
+                    boost::program_options::value<decltype(m_update)>(&m_update)->implicit_value("all")->required(),
+                    "Update the persistent images and/or the Satellite controller (SC) firmware image.  Valid values:\n"
+                    "  ALL   - All images will be updated\n"
+                    "  SHELL - Platform image\n"
+                    "  SC    - Satellite controller (Warning: Damage could occur to the device)\n"
+                    "  NO-BACKUP   - Backup boot remains unchanged",
+                    _isHidden)
+    , m_update("")
+{
+  m_optionsDescription.add_options()
+    ("device,d", po::value<decltype(m_device)>(&m_device), "The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest")
+    ("image", boost::program_options::value<decltype(m_image)>(&m_image)->multitoken(),  "Specifies an image to use used to update the persistent device.  Valid values:\n"
+                                                                    "  Name (and path) to the mcs image on disk\n"
+                                                                    "  Name (and path) to the xsabin image on disk")
+    ("help", po::bool_switch(&m_help), "Help to use this sub-command")
+  ;
+
+  m_optionsHidden.add_options()
+    ("flash-type", boost::program_options::value<decltype(m_flashType)>(&m_flashType),
+      "Overrides the flash mode. Use with caution.  Valid values:\n"
+      "  ospi\n"
+      "  ospi_versal")
+    ;
+}
+
+// Create a list of images that are known to exist based on given names and paths
+static std::vector<std::string>
+find_flash_image_paths(const std::vector<std::string>& image_list)
+{
+  std::vector<std::string> path_list;
+  auto installedShells = firmwareImage::getIntalledDSAs();
+
+  for (const auto& img : image_list) {
+    // Check if the passed in image is absolute path
+    if (boost::filesystem::is_regular_file(img)) {
+      if (boost::filesystem::extension(img).compare(".xsabin") != 0) {
+        std::cout << "Warning: Non-xsabin file detected. Development usage, this may damage the card\n";
+        if (!XBU::can_proceed(XBU::getForce()))
+          throw xrt_core::error(std::errc::operation_canceled);
```

Comment:
Shouldn't there be a messaging indicating that we cannot proceed and if the user wishes to to use the --force option.